### PR TITLE
remove moment.js

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
I tried removing this once before and well let's just say it didn't go according to plan...

Hopefully it goes better this time.

Removing moment.js should help reduce the initial bundle size pretty significantly which will increase initial loading performance and help with #4510.

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
Looks like cypress is timing out at 10 min (it usually takes around 3 min) so I'll bump it a little and see if the tests pass.

**Comment by VIKTORVAV99** [comment]
@tonypls I don't know why the CI jobs are timing out but the code should be fine.

Do you have any idea what it could be?

If not I think it's bet just to wait a little and try running the CI suite again.

**Comment by VIKTORVAV99** [comment]
It looks like it stalls here:
image

**Comment by tonypls** [comment]
Seems super slow, the server instances seem fine. I'll push to a different branch soon and see if it's suffering the same.

**Comment by tonypls** [comment]
Running cypress locally is also very hit and miss. We should look into this.

**Comment by VIKTORVAV99** [comment]
> Seems super slow, the server instances seem fine. I'll push to a different branch soon and see if it's suffering the same.

> Running cypress locally is also very hit and miss. We should look into this.

Yeah and removing moment.js itself should have no effect on cypress so I really have no idea why it stalls. We should try and push it to a local branch in this repo instead of my fork and see how that behaves but it's really strange.

I had some timeout issue for the python tests in #4528 as well so it may be indicative of a bigger issue or infrastructure change on GitHub's side.

**Comment by VIKTORVAV99** [comment]
@tonypls I commented out the lines that caused the stall both in the CI and locally I hope this is acceptable for now at least. I also set the language explicitly on each visit cause it was trying to compare the Swedish language strings with the expected English strings locally. 😅

This should be ready to be reviewed but could you just try and open up the wind/solar layers locally with live data when reviewing this, that is caused the issues in the past and I can't test it as there is no wind and solar mockdata.

**Comment by tonypls** [comment]
I'm still confused why this was failing, I don't think commenting out the tests really solves much 😆 

Could we try on a new branch, push a single line comment to trigger CI to run cypress then add just the moment changes and see if the tests still pass?

**Comment by VIKTORVAV99** [comment]
> I'm still confused why this was failing, I don't think commenting out the tests really solves much 😆
> 
> Could we try on a new branch, push a single line comment to trigger CI to run cypress then add just the moment changes and see if the tests still pass?

I think it was failing because the DOM elements where never mounted properly (the test code already mentions this) so when trying to click on the element (that did not exist in DOM) the tests simply stalled.

I'm not totally sure if that will work due to the caching in the `cypress-io/github-action` but we can certainly try.

**Comment by VIKTORVAV99** [comment]
> > > I'm still confused why this was failing, I don't think commenting out the tests really solves much 😆
> > > Could we try on a new branch, push a single line comment to trigger CI to run cypress then add just the moment changes and see if the tests still pass?
> > 
> > 
> > I think it was failing because the DOM elements where never mounted properly (the test code already mentions this) so when trying to click on the element (that did not exist in DOM) the tests simply stalled.
> > I'm not totally sure if that will work due to the caching in the `cypress-io/github-action` but we can certainly try.
> 
> Ah okay, so the cypress tests are cached and don't run on each CI push?

Internally the cypress action has at least 2 caches, but I don't know how and what they are set up to cache. I know they cache the `node_modules` (which is probably why it fails in this PR since that got invalidated) but I don't know what else they cache.

**Comment by VIKTORVAV99** [comment]
I can try a PR that just updates cypress to 9.7, that should in theory invalidate all the caches (IMO).

**Comment by tonypls** [comment]
Yeah, that would be a good test 👍 

If that fails then I think we'll need to do an isolated dive into how to make the cypress suit more reliable

**Comment by tonypls** [comment]
Let's soldier on, could you bring the timeout back down though?

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ jobs:
   cypress-test:
     name: Cypress tests
     runs-on: ubuntu-latest
-    timeout-minutes: 10
+    timeout-minutes: 20
```
[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply .github/workflows/cypress.yml:7]: Oh that one... 😆

Yeah sure!

File: web/src/helpers/gfs.js [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
 import { protectedJsonRequest } from './api';
 
 const GFS_STEP_ORIGIN = 6; // hours
 const GFS_STEP_HORIZON = 1; // hours
 
 export function getGfsTargetTimeBefore(datetime) {
-  let horizon = moment(datetime).utc().startOf('hour');
-  while (horizon.hour() % GFS_STEP_HORIZON !== 0) {
-    horizon = horizon.subtract(1, 'hour');
+  let horizon = startOfHour(datetime);
```
Comment (tonypls) [inline web/src/helpers/gfs.js:9]: I think we need to add
``` let horizon = startOfHour(datetime ? datetime : new Date());```
here

Comment (VIKTORVAV99) [reply web/src/helpers/gfs.js:9]: Wouldn't `let horizon = startOfHour(datetime ?? new Date())` make more sense if that's the case?
Assuming datetime is either undefined or null and not explicitly false.

Comment (tonypls) [reply web/src/helpers/gfs.js:9]: yeah sounds good. I think this could have been something to do with the tests not being happy for the layers.

Comment (VIKTORVAV99) [reply web/src/helpers/gfs.js:9]: I'll add the nullish coalescing and then add back the tests and see if they work now.

**Pull request merged**